index.js

```diff
@@ -205,3 +206,203 @@ export function checkDataIntegrityProofFormat({
+        });
+        it('If the "proof.created" field is missing, a ' +
+          '"MALFORMED_PROOF_ERROR" MUST be raised.', async function() {
+          this.test.cell = {columnId: vendorName, rowId: this.test.title};
+          // FIXME: Fix test to check if a cryptographic suite requires the
+          // “proof.created” value, and if it is not set, a
+          // “MALFORMED_PROOF_ERROR” MUST be raised.
+          const credential = credentials.clone('noCreated');
+          await verificationFail({credential, verifier});
+        });
+        it('If the "proof.created" field is invalid, a ' +
+          '"MALFORMED_PROOF_ERROR" MUST be raised.', async function() {
+          this.test.cell = {columnId: vendorName, rowId: this.test.title};
+          // FIXME: Fix test to check if a cryptographic suite requires the
+          // “proof.created” value
+          const credential = credentials.clone('invalidCreated');
+          await verificationFail({credential, verifier});
+        });
+        it('If the "proof.created" field is set and it deviates more than ' +
+          '"options.acceptableCreatedTimeDeviationInSeconds" seconds, a ' +
+          '"CREATED_TIME_DEVIATION_ERROR" MUST be raised.', async function() {
+          this.test.cell = {columnId: vendorName, rowId: this.test.title};
+          // FIXME: Fix test to check if a cryptographic suite requires the
+          // “proof.created” value
+          const credential = credentials.clone('issuedVc');
+          // intentionally set the created date to be a week ago
+          const created = new Date();
+          created.setDate(created.getDate() - 7);
+          credential.proof.created =
+            created.toISOString().replace(/\.\d+Z$/, 'Z');
+          await verificationFail({
+            credential, verifier, options: {
+              // set acceptable created time deviation to one day => 86400s
+              acceptableCreatedTimeDeviationInSeconds: 86400
+            }
+          });
+        });
+        it('If the "proof.proofValue" field is not a multibase-encoded ' +
+          'base58-btc value, an "INVALID_PROOF_VALUE" error MUST be raised.',
+        async function() {
+          this.test.cell = {columnId: vendorName, rowId: this.test.title};
+          const credential = credentials.clone('issuedVc');
+          // remove the initial z
+          credential.proof.proofValue = credential.proof.proofValue.slice(1);
+          await verificationFail({credential, verifier});
+        });
+        it('If the "options.domain" is set and it does not match ' +
+          '"proof.domain", an "INVALID_DOMAIN_ERROR" MUST be raised.',
+        async function() {
+          this.test.cell = {columnId: vendorName, rowId: this.test.title};
+          const credential = credentials.clone('issuedVc');
+          // intentionally set the domain to be different from the expected
+          // domain in options
+          credential.proof.domain = 'not-domain.example';
```

Comment:
we might need a valid signed VC with an invalid domain. modifying the proof of a valid vc will cause verification to fail so can we issue a VC with a domain set to something that can not be a domain name? so for instance issue a vc with the domain `vc.domain.test.invalid ` and make a note that https://en.wikipedia.org/wiki/.invalid#:~:text=The%20name%20invalid%20is%20reserved,(DNS)%20of%20the%20Internet.

`.invalid` can not never be a valid domain name.


looks like your existing example domain will also be invalid: https://en.wikipedia.org/wiki/Example.com

http://www.example.com/

so `invalid-vc-domain.example.com` would work.

Reply:
Thanks, I've updated the test to use issued VC with an invalid domain [9ba699a](https://github.com/w3c-ccg/data-integrity-test-suite-assertion/pull/13/commits/9ba699a076483b3e155d31cae9559ab780145949)

Reply:
awesome and thanks!